src/badthreads.h

```diff
@@ -0,0 +1,58 @@
+#ifndef _BADTHREADS_H_
+#define _BADTHREADS_H_
+
+/*
+ * This file contains functions and symbols that are defined in C11 threads.h.
+ * If any of these symbols are used in a file that includes badthreads.h, it
+ * should throw an error at compile time.
+ *
+ * The purpose of this file is to make sure that code does not accidentally
+ * use symbols from threads.h. If this happens, and the system C library has
+ * C11-style thread support, then the resulting object could link to the
+ * system's functions that have the same name, instead of the local functions.
+*/
+
+#define thrd_t              THREADS_H_ERROR
```

Comment:
This name (`thrd_t`) still exists even though it's listed here as bad. Pretty sure that means that `THREADS_H_ERROR` becomes the name of the type, which does mean the wrong type name is avoided, but I'm not sure that's what was intended.